src/System.ComponentModel.TypeConverter/src/System/ComponentModel/TypeDescriptor.cs

```diff
@@ -2637,6 +2724,180 @@ public static void SortDescriptorArray(IList infos)
         }
 
         /// <summary>
+        ///     This class is a type description provider that works with the IComNativeDescriptorHandler
+        ///     interface.
+        //// </summary>
+        private sealed class ComNativeDescriptionProvider : TypeDescriptionProvider
+        {
+#pragma warning disable 618
+            private IComNativeDescriptorHandler _handler;
+
+            internal ComNativeDescriptionProvider(IComNativeDescriptorHandler handler)
+            {
+                _handler = handler;
+            }
+
+            /// <summary>
+            ///     Returns the COM handler object.
+            //// </summary>
+            internal IComNativeDescriptorHandler Handler
+            {
+                get
+                {
+                    return _handler;
+                }
+                set
+                {
+                    _handler = value;
+                }
+            }
+#pragma warning restore 618
+            
+            /// <summary>
+            ///     Implements GetTypeDescriptor.  This creates a custom type
+            ///     descriptor that walks the linked list for each of its calls.
+            //// </summary>
+            
+            [SuppressMessage("Microsoft.Globalization", "CA1303:DoNotPassLiteralsAsLocalizedParameters")]
+            public override ICustomTypeDescriptor GetTypeDescriptor(Type objectType, object instance)
+            {
+                if (objectType == null)
+                {
+                    throw new ArgumentNullException("objectType");
+                }
+
+                if (instance == null)
+                {
+                    return null;
+                }
+
+                if (!objectType.IsInstanceOfType(instance))
+                {
+                    throw new ArgumentException("instance");
+                }
+
+                return new ComNativeTypeDescriptor(_handler, instance);
+            }
+
+            /// <summary>
+            ///     This type descriptor sits on top of a native
+            ///     descriptor handler.
+            //// </summary>
+            private sealed class ComNativeTypeDescriptor : ICustomTypeDescriptor
+            {
+#pragma warning disable 618
+                private IComNativeDescriptorHandler _handler;
+                private object _instance;
+
+                /// <summary>
+                ///     Creates a new ComNativeTypeDescriptor.
+                //// </summary>
+                internal ComNativeTypeDescriptor(IComNativeDescriptorHandler handler, object instance)
+                {
+                    _handler = handler;
+                    _instance = instance;
+                }
+#pragma warning restore 618
+
+                /// <summary>
+                ///     ICustomTypeDescriptor implementation.
+                //// </summary>
```

Comment:
Nit: All these `<summary>ICustomTypeDescriptor implementation.</summary>` don't really provide much value. Could they all be deleted?